database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java

```diff
@@ -90,6 +110,17 @@ public void cleanup() {
         mSnapshots.removeChangeEventListener(this);
     }
 
+    @SuppressWarnings("unused")
+    @OnLifecycleEvent(Lifecycle.Event.ON_ANY)
+    void cleanup(LifecycleOwner source, Lifecycle.Event event) {
+        if (event == Lifecycle.Event.ON_STOP) {
+            cleanup();
+            notifyDataSetChanged();
```

Comment:
Without this I was getting `Inconsistency detected. Invalid view holder adapter positionViewHolder` exceptions. I also got worried about the RecyclerView querying the `FirebaseArray` while it still had another listener and still getting the items so I added [the check](https://github.com/firebase/FirebaseUI-Android/pull/827/files#diff-724a30131c3284a3a9d723d6bb38561dR168) in `getItemCount`.